src/main/scala/io/smartdatalake/workflow/action/DeduplicateAction.scala

```diff
@@ -93,26 +93,55 @@ case class DeduplicateAction(override val id: ActionObjectId,
     val existingDf = if (output.isTableExisting) {
       Some(output.getDataFrame())
     } else None
-    val deduplicateTransformer = deduplicateDataFrame(existingDf, pks, timestamp) _
+    val deduplicateTransformer = DeduplicateAction.deduplicateDataFrame(existingDf, pks, timestamp, ignoreOldDeletedColumns, ignoreOldDeletedNestedColumns) _
     applyTransformations(subFeed, transformer, columnBlacklist, columnWhitelist, additionalColumns, standardizeDatatypes, Seq(deduplicateTransformer), filterClauseExpr)
   }
 
   /**
-   * deduplicates a SubFeed.
+   * @inheritdoc
    */
-  private def deduplicateDataFrame(existingDf: Option[DataFrame], pks: Seq[String], refTimestamp: LocalDateTime)(df: DataFrame)(implicit session: SparkSession): DataFrame = {
+  override def factory: FromConfigFactory[Action] = DeduplicateAction
+}
+
+object DeduplicateAction extends FromConfigFactory[Action] {
 
+  /**
+   * @inheritdoc
+   */
+  override def fromConfig(config: Config, instanceRegistry: InstanceRegistry): DeduplicateAction = {
+    import configs.syntax.ConfigOps
+    import io.smartdatalake.config._
+    implicit val instanceRegistryImpl: InstanceRegistry = instanceRegistry
+    config.extract[DeduplicateAction].value
+  }
+
+  /**
+   * deduplicates a SubFeed.
+   */
+  def deduplicateDataFrame(existingDf: Option[DataFrame], pks: Seq[String], refTimestamp: LocalDateTime, ignoreOldDeletedColumns: Boolean, ignoreOldDeletedNestedColumns: Boolean)(df: DataFrame)(implicit session: SparkSession): DataFrame = {
     // enhance
     val enhancedDf = df.withColumn(TechnicalTableColumn.captured.toString, ActionHelper.ts1(refTimestamp))
 
     // deduplicate
     if (existingDf.isDefined) {
       // apply schema evolution
       val (baseDf, newDf) = SchemaEvolution.process(existingDf.get, enhancedDf, ignoreOldDeletedColumns = ignoreOldDeletedColumns, ignoreOldDeletedNestedColumns = ignoreOldDeletedNestedColumns)
-      deduplicate(baseDf, newDf, pks)
+      // Schema evolution puts new columns at the end of the column list, so we need to move the technical captured column back to the end
+      deduplicate(moveCapturedColumnToEnd(baseDf), moveCapturedColumnToEnd(newDf), pks)
     } else enhancedDf
   }
 
+  /**
+   * The techincal "captured" column needs to be the last column in the DataFrame so that the union in the deduplicate method words
+   * In Spark, unions do not work by column name but by column index, so the column order of the dataframes to be unioned needs to be equal
+   * @param df A DataFrame
+   * @return A DataFrame where the techinical "captured" column is the last column
+   */
+  def moveCapturedColumnToEnd(df: DataFrame) : DataFrame = {
```

Comment:
I also came accross this problem when working on #182. There is a better solution by using df.unionByName in deduplicate(...). Shouldn't we go directly for that?